pyteal/compiler/compiler.py

```diff
@@ -20,6 +20,9 @@ def __init__(
     ):
         self.mode = mode
         self.version = version
+        self.currentLoop: Optional[Expr] = None
```

Comment:
Should these be in `CompileOptions`? Or perhaps we can create another class for them, but I'm personally not sure if/how it will be affected when passing it around in the `__teal__` methods.

Reply:
We decided to put these variables on `CompileOptions` because that class is essentially the de facto context manager during compilation. Maybe we can improve on this in the future, but I think it's ok for now